### PR TITLE
Add registries & replications client, licensing information

### PR DESCRIPTION
**Pull request by elenz97**
Add upstream typings for registries and replications as well as a coverage of the basic API routes.

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -124,12 +124,34 @@ func (c *Client) Users() *UserClient {
...
+}
+
+func (c *Client) Registries() *RegistryClient {
+	r := &RegistryClient{Client: &Client{
+		baseURL:  c.baseURL,
+		username: c.username,
+		password: c.password,
+	}}
+
+	r.baseURLSuffix = "registries"
```
Comment (martin-helmich) [inline client.go:144]: This seems to be a repeating pattern. Can we maybe extract the "copy client with separate base URL" into its own method?

```go
func (c *Client) withBaseURL(suffix string) *Client {
	return &Client{
		baseURL:  c.baseURL,
		username: c.username,
		password: c.password,
		baseURLSuffix: suffix,
	}
}
```

File: replication_types.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	Status     string      `orm:"column(status)" json:"status"`
+	StatusText string      `orm:"column(status_text)" json:"status_text"`
+	Total      int         `orm:"column(total)" json:"total"`
+	Failed     int         `orm:"column(failed)" json:"failed"`
+	Succeed    int         `orm:"column(succeed)" json:"succeed"`
+	InProgress int         `orm:"column(in_progress)" json:"in_progress"`
+	Stopped    int         `orm:"column(stopped)" json:"stopped"`
+	Trigger    TriggerType `orm:"column(trigger)" json:"trigger"`
+	StartTime  time.Time   `orm:"column(start_time)" json:"start_time"`
+	EndTime    time.Time   `orm:"column(end_time)" json:"end_time"`
```
Comment (martin-helmich) [inline replication_types.go:18]: What are the `orm` annotations for? Or is this copy&pasted from somewhere else (if so, please provide attribution)?

File: client.go (hunk 1 above)
Comment (elenz97) [reply client.go:144]: I agree with replacing and removing some clutter. Before committing, would you be fine with this?
```go
func (c *Client) withURLSuffix(suffix string) *Client {
	return &Client{
		baseURL:       c.baseURL,
		username:      c.username,
		password:      c.password,
		baseURLSuffix: suffix,
	}
}

func (c *Client) Projects() *ProjectClient {
	return &ProjectClient{c.withURLSuffix("projects")}
}

[...]
```

Comment (martin-helmich) [reply client.go:144]: Exactly what I had in mind :+1:

File: project.go [hunk 3]
```diff
@@ -36,7 +37,8 @@ func (s *ProjectClient) CheckProject(projectName string) error {
 // Return specific project details
 func (s *ProjectClient) GetProjectByID(pid int64) (Project, error) {
 	var project Project
-	resp, _, errs := s.NewRequest(gorequest.GET, "/"+I64toA(pid)).
+	resp, _, errs := s.NewRequest(gorequest.GET,
+		url.PathEscape("/"+I64toA(pid))).
```
Comment (martin-helmich) [inline project.go:41]: Caution! ⚠️  Now, you will also escape the `"/"` (to `%2F`, most probably).

Comment (martin-helmich) [reply project.go:41]: Integer parameters should be safe, anyway.

**Pull request merged**